src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/mappers/RuleEntryConverter.java

```diff
@@ -0,0 +1,247 @@
+import org.openkilda.messaging.info.switches.v2.RuleInfoEntryV2.WriteMetadata;
+import org.openkilda.messaging.info.switches.v2.action.BaseAction;
+import org.openkilda.messaging.info.switches.v2.action.CopyFieldActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.GroupActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.MeterActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.PopVlanActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.PopVxlanActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.PortOutActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.PushVlanActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.PushVxlanActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.SetFieldActionEntry;
+import org.openkilda.messaging.info.switches.v2.action.SwapFieldActionEntry;
+import org.openkilda.model.GroupId;
+import org.openkilda.model.IPv4Address;
+import org.openkilda.model.MeterId;
+import org.openkilda.model.cookie.CookieBase;
+import org.openkilda.rulemanager.Field;
+import org.openkilda.rulemanager.FlowSpeakerData;
+import org.openkilda.rulemanager.Instructions;
+import org.openkilda.rulemanager.OfFlowFlag;
+import org.openkilda.rulemanager.OfMetadata;
+import org.openkilda.rulemanager.OfTable;
+import org.openkilda.rulemanager.ProtoConstants;
+import org.openkilda.rulemanager.ProtoConstants.PortNumber.SpecialPortType;
+import org.openkilda.rulemanager.action.Action;
+import org.openkilda.rulemanager.action.GroupAction;
+import org.openkilda.rulemanager.action.MeterAction;
+import org.openkilda.rulemanager.action.PortOutAction;
+import org.openkilda.rulemanager.action.PushVxlanAction;
+import org.openkilda.rulemanager.action.SetFieldAction;
+import org.openkilda.rulemanager.action.SwapFieldAction;
+import org.openkilda.rulemanager.action.noviflow.CopyFieldAction;
+import org.openkilda.rulemanager.action.noviflow.OpenFlowOxms;
+import org.openkilda.rulemanager.match.FieldMatch;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Mapper
+public class RuleEntryConverter {
+    public static final RuleEntryConverter INSTANCE = Mappers.getMapper(RuleEntryConverter.class);
+
+    /**
+     * Converts rule representation.
+     */
+    public RuleInfoEntryV2 toRuleEntry(FlowSpeakerData speakerData) {
+        return RuleInfoEntryV2.builder()
+                .cookie(Optional.ofNullable(speakerData.getCookie())
+                        .map(CookieBase::getValue)
+                        .orElse(null))
+                .priority(speakerData.getPriority())
+                .tableId(Optional.ofNullable(speakerData.getTable())
+                        .map(OfTable::getTableId)
+                        .orElse(null))
+                .cookieKind("TBD")
```

Comment:
If we do not have it we should set it to null and add `todo` to do not return "TBD" from out API

Reply:
will do, thanks :)